src/classes/LVL_LevelAssign_BATCH.cls

```diff
@@ -136,7 +138,12 @@ public without sharing class LVL_LevelAssign_BATCH implements Database.Batchable
                         break;
                     }
 
-                } // levels               
+                } // levels
+                // After looping through all levels in the current ladder if no match found but level set clear level
+                if (!matchedLvl && lvlIdPrev != null) {
+                    sobj.put(strLadder, null);
+                    mapSObjUpdate.put(sobj.Id, sobj);
+                }
```

Comment:
all looks good, except when we clear out the current level, we should store that level in the previous level on the object.